api/v1beta1/gslb_types.go

```diff
@@ -45,6 +45,8 @@ type Strategy struct {
 type ResourceRef struct {
 	// Ingress selects a kubernetes.networking.k8s.io/v1.Ingress resource
 	Ingress metav1.LabelSelector `json:"ingress,omitempty"`
+	// IstioVirtualService selects a networking.istio.io/v1.VirtualService resource
+	IstioVirtualService metav1.LabelSelector `json:"istioVirtualService,omitempty"`
```

Comment:
Now I am slightly concerned that we have to change/extend the API everytime we add new integration.

Is it possible to make resourceRef more dynamic and clear, something like

```
  resourceRef:
      kind: Ingress
      apiVersion: networking.k8s.io/v1
      matchLabels:
        app: roundrobin-test
```

```
  resourceRef:
      kind: VirtualService
      apiVersion: networking.istio.io/v1beta1
      matchLabels:
        app: roundrobin-test
```

It will make obvious what kind of exact gvk we are selecting and will keep the main Gslb CRD stable.

Reply:
Good suggestion. We can definitely do that and keep the CRD stable